opslevel/resource_opslevel_check_tool_usage.go

```diff
@@ -163,14 +179,38 @@ func (r *CheckToolUsageResource) Create(ctx context.Context, req resource.Create
 	}
 
 	input.ToolCategory = opslevel.ToolCategory(planModel.ToolCategory.ValueString())
-	if planModel.ToolNamePredicate != nil {
-		input.ToolNamePredicate = planModel.ToolNamePredicate.ToCreateInput()
+
+	// convert environment_predicate object to model from plan
```

Comment:
After reading this `Create()` function I don't see any cases where `resp.Diagnostics` is actually checked